sdk/cognitiveservices/cognitiveservices-textanalytics/review/cognitiveservices-textanalytics.api.md

```diff
@@ -0,0 +1,325 @@
+    erroneousDocumentCount: number;
+    transactionCount: number;
+    validDocumentCount: number;
+}
+
+// @public
+export interface SentenceSentiment {
+    length: number;
+    offset: number;
+    sentenceScores: SentimentConfidenceScorePerLabel;
+    sentiment: SentenceSentimentValue;
+    warnings?: string[];
+}
+
+// @public
+export type SentenceSentimentValue = 'positive' | 'neutral' | 'negative';
+
+// @public
+export interface SentimentConfidenceScorePerLabel {
+    // (undocumented)
+    negative: number;
+    // (undocumented)
+    neutral: number;
+    // (undocumented)
+    positive: number;
+}
+
+// @public
+export class TextAnalyticsClient {
+    constructor(endpointUrl: string, credential: TokenCredential | CognitiveServicesCredentials, options?: TextAnalyticsClientOptions);
+    // (undocumented)
+    analyzeSentiment(input: string[], language?: string, options?: AnalyzeSentimentOptions): Promise<AnalyzeSentimentResultCollection>;
+    // (undocumented)
+    analyzeSentiment(input: MultiLanguageInput[], options?: AnalyzeSentimentOptions): Promise<AnalyzeSentimentResultCollection>;
+    defaultCountryHint: string;
+    defaultLanguage: string;
+    // (undocumented)
+    detectLanguage(input: string[], countryHint?: string, options?: DetectLanguagesOptions): Promise<DetectLanguageResultCollection>;
+    // (undocumented)
+    detectLanguage(input: LanguageInput[], options?: DetectLanguagesOptions): Promise<DetectLanguageResultCollection>;
+    readonly endpointUrl: string;
+    // (undocumented)
+    extractEntityLinking(input: string[], language?: string, options?: ExtractEntityLinkingOptions): Promise<ExtractLinkedEntitiesResultCollection>;
+    // (undocumented)
+    extractEntityLinking(input: MultiLanguageInput[], options?: ExtractEntityLinkingOptions): Promise<ExtractLinkedEntitiesResultCollection>;
+    // (undocumented)
+    extractKeyPhrases(input: string[], language?: string, options?: ExtractKeyPhrasesOptions): Promise<ExtractKeyPhrasesResultCollection>;
+    // (undocumented)
+    extractKeyPhrases(input: MultiLanguageInput[], options?: ExtractKeyPhrasesOptions): Promise<ExtractKeyPhrasesResultCollection>;
+    // (undocumented)
+    recognizeEntities(input: string[], language?: string, options?: RecognizeEntitiesOptions): Promise<RecognizeEntitiesResultCollection>;
+    // (undocumented)
+    recognizeEntities(input: MultiLanguageInput[], options?: RecognizeEntitiesOptions): Promise<RecognizeEntitiesResultCollection>;
+    // (undocumented)
+    recognizePiiEntities(input: string[], language?: string, options?: RecognizePiiEntitiesOptions): Promise<RecognizeEntitiesResultCollection>;
+    // (undocumented)
+    recognizePiiEntities(input: MultiLanguageInput[], options?: RecognizePiiEntitiesOptions): Promise<RecognizeEntitiesResultCollection>;
+    // (undocumented)
+    singleAnalyzeSentiment(inputText: string, language?: string, options?: AnalyzeSentimentOptions): Promise<AnalyzeSentimentSuccessResult>;
```

Comment:
is the naming `singleXxxx` consistent across languages? I feel it strange for a function to not start with a verb.

Reply:
@bterlson is also not a fan. This pattern came from Python, but they put it on their namespace rather than their client. I'm very open to suggestions about naming here.

Reply:
Had a convo with @xirzec and Peter about this. I think the `single` prefixed methods are my least favorite of the possible solutions in this space. For clarity, we have:

1. `single` prefix
2. separate batch and single clients
3. overloads for single item and batch (with different throwing behavior for each case)
4. top-level exports for single items
5. only offering batching

I think option 5 is my favorite at this point - wrapping in arrays isn't too hard for JS users, and if we find that this is too troublesome, we can pretty quickly deliver any of the other options above.

Any objections to rolling with just batching for now?

Reply:
What's the different throwing behavior?  If we do overloads now but internally use batch for both, is it a the concern that we might want to bring single throwing behavior in thus would introduce breaking behavior

Reply:
Single methods throw on error, batch methods do not throw as there can be a mix of success and errrors.